compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -616,37 +617,28 @@ populateConfigInfo(const llvm::SetVector<linalg::LinalgOp> &computeOps,
     if (hasExternalCapture(cast<linalg::GenericOp>(linalgOp))) {
       return false;
     }
-    // If some of the users are in computeOps and some are outside of
-    // computeOps; attach lowering config, since the op can't be fused.
-    if (llvm::any_of(linalgOp->getUsers(),
-                     [&](Operation *user) {
-                       auto linalgUser = dyn_cast<linalg::LinalgOp>(user);
-                       return linalgUser && computeOps.contains(linalgUser);
-                     }) &&
-        llvm::any_of(linalgOp->getUsers(), [&](Operation *user) {
+    // We want to attach a lowering config to this operation if it introduces
+    // a new dimension, when going by topological order in the backward slice.
+    // The only two ways to introduce a new dimension are:
+    //
+    // 1. We have a reduction dimension.
+    if (hasReductionIterator(linalgOp)) {
+      return true;
+    }
+    // 2. There is no consumer which is a compute op (i.e. it already
+    // has some way of getting fused).
+    if (!llvm::any_of(linalgOp->getUsers(), [&](Operation *user) {
```

Comment:
nit/optional: use `llvm::none_of`?

Reply:
none_of returns false if the list is empty, while !any_of returns true if the list is empty. I want the true behavior if the list is empty.